addons/mail/static/src/new/core/thread_model.js

```diff
@@ -122,15 +122,28 @@ export class Thread {
         return Math.max(...this.messages);
     }
 
+    get mostRecentNonTransientMessage() {
+        if (this.messages.length === 0) {
+            return undefined;
+        }
+        const oldestNonTransientMessageId = [...this.messages]
+            .reverse()
+            .find((messageId) => Number.isInteger(messageId));
```

Comment:
Only if we can guarantee the messages are always sorted then it's fine to have this. For the rest, it looks good!

Reply:
Yes this code assumes `this.messages` is always ordered.
I think it's good to make this assumption, as this makes code more performant.
If this becomes wrong or an issue in future, then only this part of code fails and we can just fix it.